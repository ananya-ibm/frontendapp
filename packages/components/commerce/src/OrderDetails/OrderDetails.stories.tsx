/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

import React from 'react';
import { OrderDetails } from './OrderDetails';

export default {
  title: 'Components/Commerce/OrderDetails',
  component: OrderDetails
};

export const normal = args => <OrderDetails {...args} />;
normal.args = {
  title: 'Order Details',
  subTitle: 'Order Number: ',
  orderNumber: '0003835',
  orderDate: 'September 15, 2018',
  billingAddress: 'John Smith, 640, Hamilton circle Malvern, PA 19355',
  shippingAddress: 'John Smith, 640, Hamilton circle Malvern, PA 19355',
  paymentMethod: 'visa********9257',
  orderStatus: 'Shipped on Tuesday, June 23 USPS Priority Mail',
  trackingId: '097654334566111222221111',
  subTotalPrice: {
    value: '59.99',
    currency: 'USD',
    format: 'en-US'
  },
  totalItems: 3,
  discountPrice: {
    prefix: '-',
    value: '10.99',
    currency: 'USD',
    format: 'en-US'
  },
  shippingCharge: {
    value: '5.50',
    currency: 'USD',
    format: 'en-US'
  },
  taxCharges: {
    value: '2.80',
    currency: 'USD',
    format: 'en-US'
  },
  totalDays: 2,
  grandTotal: {
    value: '57.30',
    currency: 'USD',
    format: 'en-US'
  },
  orders: [
    {
      id: '0002',
      quantity: 1,
      brandName: 'Titan',
      size: 'Large',
      color: 'Golden',
      item: {
        thumbnail: 'https://images.unsplash.com/photo-1476286768413-e7051cdb2179',
        name: 'Titan Raga'
      },
      linePrice: {
        value: '9.99',
        currency: 'USD',
        format: 'en-US'
      }
    },
    {
      id: '0003',
      quantity: 2,
      brandName: 'Rebook',
      size: '7',
      color: 'Black',
      item: {
        thumbnail: 'https://images.unsplash.com/photo-1596018589855-e9a2a91f687f',
        name: 'Athletic Shoes'
      },
      linePrice: {
        value: '24.54',
        currency: 'USD',
        format: 'en-US'
      }
    }
  ]
};
