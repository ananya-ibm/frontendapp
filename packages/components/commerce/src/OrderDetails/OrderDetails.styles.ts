/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

import styled from 'styled-components';
import theme from './OrderDetails.theme';
import { ImageTile } from '../ImageTile/ImageTile';

export const OrderDetails = styled('div')`
  padding: ${props => theme(props).padding};

  & .color {
    padding-left: 0;
    padding-right: 0;
  }

  & .orderNumber {
    padding-left: ${props => theme(props).margin};
  }

  & .description {
    margin-top: ${props => theme(props).marginSmall};
  }

  & .priceTotal {
    padding-top: ${props => theme(props).marginExtraLarge};
  }

  & .priceRow {
    padding-left: ${props => theme(props).marginExtraLarge};
  }
`;

export const Heading = styled('h3')`
  font-weight: 700;
`;

export const SubHeading = styled('h4')`
  font-weight: 700;
`;

export const Field = styled('div')`
  margin-top: ${props => theme(props).marginMedium};
`;

export const BoldText = styled('div')`
  font-weight: 700;
`;

export const OrderHeader = styled('div')`
  background-color: ${props => props.theme.colors.backgrounds.panels.primary.base};
  border: 0.1rem solid ${props => props.theme.colors.backgrounds.panels.primary.base};
  border-bottom: 0.1rem solid ${props => theme(props).border};
  margin-top: ${props => theme(props).marginMedium};
  padding: ${props => theme(props).marginLarge} ${props => theme(props).marginMedium};
`;

export const Text = styled('div')`
  font-size: 0.8rem;
  line-height: 1.4;
`;

export const OrderBody = styled('div')`
  border: 0.1rem solid ${props => theme(props).border};
  border-top: none;
  padding: ${props => theme(props).marginMedium};
`;

export const Thumbnail = styled(ImageTile)`
  max-width: 100%;
`;
