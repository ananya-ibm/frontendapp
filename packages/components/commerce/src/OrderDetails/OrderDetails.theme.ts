/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

type Props = {
  padding: string;
  marginMedium: string;
  border: string;
  margin: string;
  marginLarge: string;
  marginSmall: string;
  marginExtraLarge: string;
};

declare global {
  interface EXOComponentStyles {
    commerce_orderDetails?: Partial<Props>;
  }
}

export default (props: EXOThemeProps): Props => {
  return {
    // Default theme
    padding: props.theme.spacing.inline.s5,
    marginMedium: props.theme.spacing.inline.s7,
    border: props.theme.colors.backgrounds.panels.primary.base,
    margin: props.theme.spacing.inline.s5,
    marginLarge: props.theme.spacing.inline.s6,
    marginSmall: props.theme.spacing.inline.s4,
    marginExtraLarge: '2.5rem',

    // Override with custom theme
    ...props.theme?.byComponent?.commerce_orderDetails
  };
};
