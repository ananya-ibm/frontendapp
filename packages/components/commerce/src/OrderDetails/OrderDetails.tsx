/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

import React from 'react';
import { Grid, Row, Column } from '@exo/frontend-components-base';
import { MonetaryAmount } from '@exo/frontend-components-core';
import { getClientImagePath } from '@exo/frontend-common-utils';
import { OrderSummary } from '../OrderSummary/OrderSummary';
import * as S from './OrderDetails.styles';

export const OrderDetails = ({
  title = 'Order Details',
  orderNumber,
  orderDate,
  billingAddress,
  shippingAddress,
  paymentMethod,
  orderStatus,
  trackingId,
  grandTotal,
  subTotalPrice,
  totalItems,
  discountPrice,
  shippingCharge,
  totalDays,
  taxCharges,
  orders
}: Props) => {
  // TODO: There's quite a bit of polish for mobile devices
  return (
    <S.OrderDetails>
      {title && <S.Heading>{title}</S.Heading>}
      <S.Field>
        <S.SubHeading>
          Order Number:<span className="orderNumber">{orderNumber}</span>
        </S.SubHeading>
      </S.Field>
      <S.Field>
        <S.Text>Date ordered:{orderDate} </S.Text>
      </S.Field>
      <Grid isFluid>
        <S.Field>
          <Row>
            <Column sm={'100%'} lg={'33%'}>
              <S.BoldText>
                <S.Text>Billing Address</S.Text>
              </S.BoldText>
              <S.Text>{billingAddress}</S.Text>
            </Column>
            <Column sm={'100%'} lg={'33%'}>
              <S.BoldText>
                <S.Text>Shipping Address</S.Text>
              </S.BoldText>
              <S.Text>{shippingAddress}</S.Text>
            </Column>
            <Column sm={'100%'} lg={'33%'}>
              <S.BoldText>
                <S.Text>Payment Method</S.Text>
              </S.BoldText>
              <S.Text>{paymentMethod}</S.Text>
            </Column>
          </Row>
        </S.Field>
        <S.OrderHeader>
          <Row>
            <Column sm={'100%'} lg={3}>
              <S.BoldText>
                <S.Text>Order Status</S.Text>
              </S.BoldText>
              {orderStatus && <S.Text>{orderStatus}</S.Text>}
            </Column>
            <Column sm={'100%'} lg={3}>
              <S.BoldText>
                <S.Text>Tracking #</S.Text>
              </S.BoldText>
              {trackingId && <S.Text>{trackingId}</S.Text>}
            </Column>
          </Row>
        </S.OrderHeader>
        {orders &&
          orders.map(li => (
            <S.OrderBody key={li.id}>
              <Row>
                <Column sm={1} md={2} lg={3}>
                  <S.Thumbnail
                    src={getClientImagePath(li.item.thumbnail)}
                    alt={`An image of ${li.item.name}`}
                  />
                </Column>
                <Column sm={1} md={3} lg={10} className="description">
                  <S.Text>{li.brandName}</S.Text>
                  <S.SubHeading>{li.item.name}</S.SubHeading>
                  <Row className="description">
                    <Column>
                      {' '}
                      <S.Text>Size: {li.size}</S.Text>
                    </Column>
                    <Column className="color">
                      {' '}
                      <S.Text>Color: {li.color}</S.Text>
                    </Column>
                  </Row>
                  <S.Text className="description">Quantity: {li.quantity}</S.Text>
                </Column>
                <Column className="priceTotal" sm={1} md={3} lg={3}>
                  <Row className="priceRow">Total Price:</Row>
                  <Row className="priceRow">
                    <S.SubHeading>
                      <MonetaryAmount
                        value={li.linePrice.value}
                        currency={li.linePrice.currency}
                        format={li.linePrice.format}
                      />
                    </S.SubHeading>
                  </Row>
                </Column>
              </Row>
            </S.OrderBody>
          ))}
        <Row>
          <Column sm='0' lg="60%"></Column>
          <Column
            sm='100%'
            lg="40%"
          >
            <S.Field>
              <OrderSummary
                subTotalPrice={subTotalPrice}
                discountPrice={discountPrice}
                totalItems={totalItems}
                shippingCharge={shippingCharge}
                totalDays={totalDays}
                taxCharges={taxCharges}
                orderTotalPrice={grandTotal}
              />
            </S.Field>
          </Column>
        </Row>
      </Grid>
    </S.OrderDetails>
  );
};

type Money = {
  value?: string;
  currency?: string;
  format?: string;
};

type Props = {
  title?: string;
  orderNumber: string;
  orderDate: string;
  billingAddress: string;
  shippingAddress: string;
  paymentMethod: string;
  orderStatus?: string;
  trackingId?: string;
  totalItems?: number;
  totalDays?: number;
  subTotalPrice: Money;
  discountPrice: Money & { prefix?: string };
  shippingCharge: Money;
  taxCharges: Money;
  grandTotal: Money;
  orders: {
    id?: string;
    quantity?: number | string;
    brandName?: string;
    size?: string;
    color?: string;
    item: {
      thumbnail?: string;
      name?: string;
    };
    linePrice: Money;
  }[];
};
