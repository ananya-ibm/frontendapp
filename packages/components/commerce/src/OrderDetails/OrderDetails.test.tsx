/*
Licensed Materials - Property of IBM
694906H
(c) Copyright IBM Corp.  2020 All Rights Reserved

US Government Users Restricted Rights - Use, duplication or disclosure restricted
by GSA ADP Schedule Contract with IBM Corp.
*/

import React from 'react';
import { renderWithTheme as render } from '@testUtils';
import { axe } from 'jest-axe';
import { OrderDetails } from './OrderDetails';

const testProps = {
  title: 'Order Details',
  subTitle: 'Order Number: ',
  orderNumber: '0003835',
  orderDate: 'September 15, 2018',
  billingAddress: 'John Smith, 640, Hamilton circle Malvern, PA 19355',
  shippingAddress: 'John Smith, 640, Hamilton circle Malvern, PA 19355',
  paymentMethod: 'visa********9257',
  orderStatus: 'Shipped on Tuesday, June 23 USPS Priority Mail',
  trackingId: '097654334566111222221111',
  subTotalPrice: {
    value: '59.99',
    currency: 'USD',
    format: 'en-US'
  },
  totalItems: 3,
  discountPrice: {
    prefix: '-',
    value: '10.99',
    currency: 'USD',
    format: 'en-US'
  },
  shippingCharge: {
    value: '5.50',
    currency: 'USD',
    format: 'en-US'
  },
  taxCharges: {
    value: '2.80',
    currency: 'USD',
    format: 'en-US'
  },
  totalDays: 2,
  grandTotal: {
    value: '57.30',
    currency: 'USD',
    format: 'en-US'
  },
  orders: [
    {
      id: '0002',
      quantity: 1,
      brandName: 'Titan',
      size: 'Large',
      color: 'Golden',
      item: {
        thumbnail: 'https://images.unsplash.com/photo-1476286768413-e7051cdb2179',
        name: 'Titan Raga'
      },
      linePrice: {
        value: '9.99',
        currency: 'USD',
        format: 'en-US'
      }
    },
    {
      id: '0003',
      quantity: 2,
      brandName: 'Rebook',
      size: '7',
      color: 'Black',
      item: {
        thumbnail: 'https://images.unsplash.com/photo-1596018589855-e9a2a91f687f',
        name: 'Athletic Shoes'
      },
      linePrice: {
        value: '24.54',
        currency: 'USD',
        format: 'en-US'
      }
    }
  ]
};

describe('<OrderDetails /> component', () => {
  test('its snapshot matches', () => {
    const { container } = render(<OrderDetails {...testProps} />);

    expect(container).toBeDefined();
    expect(container.firstChild).toMatchSnapshot();
  });
  test('it passes an axe accesibility check', async () => {
    const { container } = render(<OrderDetails {...testProps} />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
