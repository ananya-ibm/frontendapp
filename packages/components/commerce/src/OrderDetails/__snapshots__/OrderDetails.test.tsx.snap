// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderDetails /> component its snapshot matches 1`] = `
.c5 {
  max-width: var(--max-width);
  padding-left: 0;
  padding-right: 0;
  max-width: 100%;
}

.c5 .cds--row > * > .cds--grid {
  padding-left: 0;
  padding-right: 0;
}

.c12 {
  max-width: var(--max-width);
}

.c12 .cds--row > * > .cds--grid {
  padding-left: 0;
  padding-right: 0;
}

.c10 .price-list {
  color: rgba(22,22,22,0.25);
  -webkit-text-decoration: line-through;
  text-decoration: line-through;
}

.c11 .cds--grid {
  padding-left: 0;
}

.c11 .cds--row {
  margin-left: 0;
}

.c11 .priceColumn {
  text-align: right;
}

.c11 .subRows {
  margin-top: 0.75rem;
}

.c13 {
  font-weight: 700;
}

.c14 {
  font-weight: 700;
}

.c0 {
  padding: 1rem;
}

.c0 .color {
  padding-left: 0;
  padding-right: 0;
}

.c0 .orderNumber {
  padding-left: 1rem;
}

.c0 .description {
  margin-top: 0.75rem;
}

.c0 .priceTotal {
  padding-top: 2.5rem;
}

.c0 .priceRow {
  padding-left: 2.5rem;
}

.c1 {
  font-weight: 700;
}

.c3 {
  font-weight: 700;
}

.c2 {
  margin-top: 2rem;
}

.c6 {
  font-weight: 700;
}

.c7 {
  background-color: #f4f4f4;
  border: 0.1rem solid #f4f4f4;
  border-bottom: 0.1rem solid #f4f4f4;
  margin-top: 2rem;
  padding: 1.5rem 2rem;
}

.c4 {
  font-size: 0.8rem;
  line-height: 1.4;
}

.c8 {
  border: 0.1rem solid #f4f4f4;
  border-top: none;
  padding: 2rem;
}

.c9 {
  max-width: 100%;
}

<div
  class="c0"
>
  <h3
    class="c1"
  >
    Order Details
  </h3>
  <div
    class="c2"
  >
    <h4
      class="c3"
    >
      Order Number:
      <span
        class="orderNumber"
      >
        0003835
      </span>
    </h4>
  </div>
  <div
    class="c2"
  >
    <div
      class="c4"
    >
      Date ordered:
      September 15, 2018
       
    </div>
  </div>
  <div
    class="c5 cds--grid cds--grid--full-width"
  >
    <div
      class="c2"
    >
      <div
        class="cds--row"
      >
        <div
          class="cds--col-sm-4 cds--col-lg-5"
        >
          <div
            class="c6"
          >
            <div
              class="c4"
            >
              Billing Address
            </div>
          </div>
          <div
            class="c4"
          >
            John Smith, 640, Hamilton circle Malvern, PA 19355
          </div>
        </div>
        <div
          class="cds--col-sm-4 cds--col-lg-5"
        >
          <div
            class="c6"
          >
            <div
              class="c4"
            >
              Shipping Address
            </div>
          </div>
          <div
            class="c4"
          >
            John Smith, 640, Hamilton circle Malvern, PA 19355
          </div>
        </div>
        <div
          class="cds--col-sm-4 cds--col-lg-5"
        >
          <div
            class="c6"
          >
            <div
              class="c4"
            >
              Payment Method
            </div>
          </div>
          <div
            class="c4"
          >
            visa********9257
          </div>
        </div>
      </div>
    </div>
    <div
      class="c7"
    >
      <div
        class="cds--row"
      >
        <div
          class="cds--col-sm-4 cds--col-lg-3"
        >
          <div
            class="c6"
          >
            <div
              class="c4"
            >
              Order Status
            </div>
          </div>
          <div
            class="c4"
          >
            Shipped on Tuesday, June 23 USPS Priority Mail
          </div>
        </div>
        <div
          class="cds--col-sm-4 cds--col-lg-3"
        >
          <div
            class="c6"
          >
            <div
              class="c4"
            >
              Tracking #
            </div>
          </div>
          <div
            class="c4"
          >
            097654334566111222221111
          </div>
        </div>
      </div>
    </div>
    <div
      class="c8"
    >
      <div
        class="cds--row"
      >
        <div
          class="cds--col-sm-1 cds--col-md-2 cds--col-lg-3"
        >
          <img
            alt="An image of Titan Raga"
            class="c9"
            src="https://images.unsplash.com/photo-1476286768413-e7051cdb2179"
          />
        </div>
        <div
          class="description cds--col-sm-1 cds--col-md-3 cds--col-lg-10"
        >
          <div
            class="c4"
          >
            Titan
          </div>
          <h4
            class="c3"
          >
            Titan Raga
          </h4>
          <div
            class="description cds--row"
          >
            <div
              class="cds--col"
            >
               
              <div
                class="c4"
              >
                Size: 
                Large
              </div>
            </div>
            <div
              class="color cds--col"
            >
               
              <div
                class="c4"
              >
                Color: 
                Golden
              </div>
            </div>
          </div>
          <div
            class="c4 description"
          >
            Quantity: 
            1
          </div>
        </div>
        <div
          class="priceTotal cds--col-sm-1 cds--col-md-3 cds--col-lg-3"
        >
          <div
            class="priceRow cds--row"
          >
            Total Price:
          </div>
          <div
            class="priceRow cds--row"
          >
            <h4
              class="c3"
            >
              <span
                class="c10"
              >
                <span
                  class="value"
                >
                  $9.99
                </span>
              </span>
            </h4>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c8"
    >
      <div
        class="cds--row"
      >
        <div
          class="cds--col-sm-1 cds--col-md-2 cds--col-lg-3"
        >
          <img
            alt="An image of Athletic Shoes"
            class="c9"
            src="https://images.unsplash.com/photo-1596018589855-e9a2a91f687f"
          />
        </div>
        <div
          class="description cds--col-sm-1 cds--col-md-3 cds--col-lg-10"
        >
          <div
            class="c4"
          >
            Rebook
          </div>
          <h4
            class="c3"
          >
            Athletic Shoes
          </h4>
          <div
            class="description cds--row"
          >
            <div
              class="cds--col"
            >
               
              <div
                class="c4"
              >
                Size: 
                7
              </div>
            </div>
            <div
              class="color cds--col"
            >
               
              <div
                class="c4"
              >
                Color: 
                Black
              </div>
            </div>
          </div>
          <div
            class="c4 description"
          >
            Quantity: 
            2
          </div>
        </div>
        <div
          class="priceTotal cds--col-sm-1 cds--col-md-3 cds--col-lg-3"
        >
          <div
            class="priceRow cds--row"
          >
            Total Price:
          </div>
          <div
            class="priceRow cds--row"
          >
            <h4
              class="c3"
            >
              <span
                class="c10"
              >
                <span
                  class="value"
                >
                  $24.54
                </span>
              </span>
            </h4>
          </div>
        </div>
      </div>
    </div>
    <div
      class="cds--row"
    >
      <div
        class="cds--col-sm-0 cds--col-lg-10"
      />
      <div
        class="cds--col-sm-4 cds--col-lg-6"
      >
        <div
          class="c2"
        >
          <div
            class="c11"
          >
            <div
              class="c12 cds--grid"
            >
              <div
                class="cds--row"
              >
                <div
                  class="cds--col-sm-4"
                >
                  <h5
                    class="c13"
                  >
                     
                    ORDER SUMMARY
                  </h5>
                  <hr
                    class="subRows"
                  />
                  <div
                    class="subRows cds--row"
                  >
                    <div
                      class="tableColumn cds--col-sm-2"
                    >
                      Subtotal
                       (3 items)
                    </div>
                    <div
                      class="priceColumn cds--col-sm-2"
                    >
                      <span
                        class="c10"
                      >
                        <span
                          class="value"
                        >
                          $59.99
                        </span>
                      </span>
                    </div>
                  </div>
                  <div
                    class="subRows cds--row"
                  >
                    <div
                      class="tableColumn cds--col-sm-2"
                    >
                      Rewards/Discounts
                    </div>
                    <div
                      class="priceColumn cds--col-sm-2"
                    >
                      <span
                        class="c10"
                      >
                        <span
                          class="prefix"
                        >
                          -
                        </span>
                        <span
                          class="value"
                        >
                          $10.99
                        </span>
                      </span>
                    </div>
                  </div>
                  <div
                    class="subRows cds--row"
                  >
                    <div
                      class="tableColumn cds--col-sm-2"
                    >
                      Shipping
                       (2 Day)
                    </div>
                    <div
                      class="priceColumn cds--col-sm-2"
                    >
                      <span
                        class="c10"
                      >
                        <span
                          class="value"
                        >
                          $5.50
                        </span>
                      </span>
                    </div>
                  </div>
                  <div
                    class="subRows cds--row"
                  >
                    <div
                      class="tableColumn cds--col-sm-2"
                    >
                      Tax
                    </div>
                    <div
                      class="priceColumn cds--col-sm-2"
                    >
                      <span
                        class="c10"
                      >
                        <span
                          class="value"
                        >
                          $2.80
                        </span>
                      </span>
                    </div>
                  </div>
                  <div
                    class="subRows cds--row"
                  >
                    <div
                      class="tableColumn cds--col-sm-2"
                    >
                      <div
                        class="c14"
                      >
                        Order Total:
                      </div>
                    </div>
                    <div
                      class="priceColumn cds--col-sm-2"
                    >
                      <div
                        class="c14"
                      >
                        <span
                          class="c10"
                        >
                          <span
                            class="value"
                          >
                            $57.30
                          </span>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
